### PR TITLE
Save a couple loads/stores in sweep pages

### PR DESCRIPTION
**Pull request by gbaraldi**
We do a load/store on every iteration, keep a temporary instead, It's a very small difference but it's visible in vtune

### CODE REVIEW

**Comment by oscardssmith** [comment]
Why the change from uint8 to uint32 (and why not uint64?)

**Comment by gbaraldi** [comment]
uint64 is probably slow on 32bit and it's annoying having to have two different codepaths, also 32bit already gets us whatever performance we could get here. Just not doing the loads every iteration already helped quite a bit.

File: src/gc.c [hunk 1]
```diff
@@ -1401,7 +1401,7 @@ static NOINLINE jl_taggedvalue_t *add_page(jl_gc_pool_t *p) JL_NOTSAFEPOINT
     jl_ptls_t ptls = jl_current_task->ptls;
     jl_gc_pagemeta_t *pg = jl_gc_alloc_page();
     pg->osize = p->osize;
-    pg->ages = (uint8_t*)malloc_s(GC_PAGE_SZ / 8 / p->osize + 1);
+    pg->ages = (uint32_t*)malloc_s(GC_PAGE_SZ / 8 / p->osize + 1);
```
Comment (vtjnash) [inline src/gc.c:1404]:
```suggestion
    pg->ages = (uint32_t*)malloc_s(LLT_ALIGN(GC_PAGE_SZ / 8 / p->osize + 1, sizeof(uint32_t)));
```

File: src/gc.c [hunk 2]
```diff
@@ -1579,17 +1580,19 @@ static jl_taggedvalue_t **sweep_page(jl_gc_pool_t *p, jl_gc_pagemeta_t *pg, jl_t
...
             msk <<= 1;
             if (!msk) {
                 msk = 1;
+                *ages = age;
                 ages++;
+                age = *ages;
             }
         }
-
+        *ages = age;
```
Comment (vtjnash) [inline src/gc.c:1595]: This might be one past the end?
```suggestion
        if (msk != 1)
            *ages = age;
```

Comment (gbaraldi) [reply src/gc.c:1595]: I guess if we hit the `if(!msk)` branch on the last iteration this is possibly not legal, but that would mean `age = *ages;` a couple lines above is also possibly illegal

Comment (vtjnash) [reply src/gc.c:1595]: Move that to the top of the for loop then?

**Pull request merged**